### DISCUSSION TITLE
More control over trix image uploads

### DISCUSSION THREAD

**Discussion by mokhosh** [filamentphp/filament#1591] [Help]
I'm trying to do these things with Trix, but I'm not sure where I should start making customisations or if there are already helpers for these:

- I want to store images in a separate folder for rich text images
- I want to disable captions for images by default, currently it shows file info if you leave it blank
- I want to resize and compress images before upload (or after if before is not possible)

**Comment by danharrin** [comment]
We have a `fileAttachmentsDirectory()` method on the Rich Editor for specifying where files are stored :)

Would welcome PRs on the other two though 👍

**Reply by mokhosh** [reply]
I'd be happy to contribute, can you please give me a general idea of where things are. I don't have a clear idea of how Filamnet is organised.

**Reply by danharrin** [reply]
All of the rich editor code will be in the `packages/forms/src/Components` and `packages/forms/resources/js` directories

**Comment by mokhosh** [comment]
Update:

Thanks for the first one.
The second one I created a pull request which seems to need more work. I'm hiding with css until that's done.

For the third one I have an issue. I couldn't do it before upload, so I'm using the `saveUploadedFileAttachmentsUsing` hook to add my resize logic before saving the file:

```php
protected function handleFileUploads(TemporaryUploadedFile $file)
{
    Image::load($file->getRealPath())
        ->fit(Manipulations::FIT_MAX, width: 1600, height: 900)
        ->optimize()
        ->save();

    return $file->storePublicly('attachments');
}
```

Now the problem I have is this stores the file directly in `attachments` folder, not in `public/attachments` as it would if I just set the folder with the method you mentioned without customising the file upload method. And there's no difference between `store` and `storePublicly` in this regard.

I tried telling it to save in `public/attachments` but then the url gets messed up and browser doesn't find the file either way.

**Comment by mokhosh** [comment]
Ok fixed it with this:
```php
return $file->store('attachments', 'public');
```
